google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -872,9 +874,14 @@ public void write(Iterable<LogEntry> logEntries, WriteOption... options) {
         logEntries =
             populateMetadata(logEntries, sharedResourceMetadata, this.getClass().getName());
       }
-      // Add partialSuccess option always for request containing instrumentation data
-      writeLogEntries(
-          logEntries, pair.x() ? Instrumentation.addPartialSuccessOption(options) : options);
+      // Add partialSuccess = true option always for request which does not have
+      // it set explicitly in options.
+      // For request containing instrumentation data (e.g. when pair.x() == true),
+      // always set or override partialSuccess with true.
+      if (pair.x() || writeOptionPartialSuccessFlag == null) {
```

Comment:
nit: I think it would be more readable if you put `shouldAddInstrumentation = pair.x();` right under `logEntries = pair.y();`, so it's clear what this is

Reply:
I believe the comment explains the meaning of `pair.x()`, so we can be good here without adding extra parameter.